src/test/client/nonblocking.cc

```diff
@@ -526,4 +526,88 @@ TEST_F(TestClient, LlreadvLlwritevZeroBytes) {
 
   client->ll_release(fh);
   ASSERT_EQ(0, client->ll_unlink(root, filename, myperm));
-}
\ No newline at end of file
+}
+
+TEST_F(TestClient, LlreadvLlwritevInvalidFileHandle) {
```

Comment:
I think we can move this test out of nonblock.cc an just use nullptr for the Context or use the sync calls?

Reply:
@vshankar I dont think so, this still tests an important code path and we need to make sure async I/O works fine for an invalid file handle. Instead I'd say let us move forward with adding a test case using sync code paths `ll_read`, `ll_write` under https://tracker.ceph.com/issues/63712. These test cases (since these sync codes are quite old) will be backportable to all squid/reef/quincy

Reply:
How about splitting the test for sync and async case then?

Reply:
I know what. What I'm saying is for the sync case, we can add test to the other source.

Reply:
> yeah that is what needs to be done under the tracker i linked above.

https://tracker.ceph.com/issues/63712